src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -268,10 +268,11 @@ public void execute(BuildResult buildResult) {
             }
 
             if (isGitHubActions(providers)) {
+                Optional<String> gitHubUrl = envVariable("GITHUB_SERVER_URL", providers);
                 Optional<String> gitHubRepository = envVariable("GITHUB_REPOSITORY", providers);
                 Optional<String> gitHubRunId = envVariable("GITHUB_RUN_ID", providers);
-                if (gitHubRepository.isPresent() && gitHubRunId.isPresent()) {
-                    buildScan.link("GitHub Actions build", "https://github.com/" + gitHubRepository.get() + "/actions/runs/" + gitHubRunId.get());
+                if (gitHubUrl.isPresent() && gitHubRepository.isPresent() && gitHubRunId.isPresent()) {
```

Comment:
Are we sure `GITHUB_SERVER_URL` is always set? I'm wondering if with some old runners or on-prem runners, it could be interesting to fall back to `github.com` if empty

Reply:
Done - if `GITHUB_SERVER_URL` is unset, `https://github.com` is now used as the default value for `gitHubUrl`.